core/embed/trezorhal/stm32f4/i2c.c

```diff
@@ -169,11 +169,25 @@ void i2c_cycle(uint16_t idx) {
 
 HAL_StatusTypeDef i2c_transmit(uint16_t idx, uint8_t addr, uint8_t *data,
                                uint16_t len, uint32_t timeout) {
-  return HAL_I2C_Master_Transmit(&i2c_handle[idx], addr, data, len, timeout);
+  HAL_StatusTypeDef ret =
+      HAL_I2C_Master_Transmit(&i2c_handle[idx], addr, data, len, timeout);
+#ifdef USE_OPTIGA
```

Comment:
seems like a bad place to put this. instead into `optiga_i2c_transmit` / receive?

Reply:
Maybe. The receive 50us delay was added intentionally to i2c driver because datasheet says (or at least i interpret it this way) that the delay needs to be there even when communicating with other devices on the i2c bus. As the transmit delay is more experiment based, it might be ok to have it only after talking to optiga, but i don't know that for sure. Does not matter much for T2B1 as there is only optiga on the bus

Reply:
if that's the case, why the optiga guards?

Reply:
The "GUARD_TIME" delay only needs to be there if optiga is on the bus. From datasheet: 
```
Note 1: For two consecutive accesses on the same device
GUARD_TIME re-specifies the value of tBUF as specified by [I2Cbus].
```
Basically what i think they are saying that i2c standard is not good enough when optiga is involved.

Reply:
that's in conflict with "the delay needs to be there even when communicating with other devices on the i2c bus"

either let's add guard time for all calls under `#ifdef USE_OPTIGA` (and add a comment explaining the above) but omit the device id check

or don't add it here but in `optiga_i2_transmit`

Reply:
moved the transmission delay to optiga transport: https://github.com/trezor/trezor-firmware/pull/3303/commits/ff3e21121dd4616c098616c1a010cf3250fa0750

leaving 50us guard_time on the i2c driver level.